assets/js/icons/icon/index.js

```diff
@@ -1,14 +1,17 @@
 /**
  * External dependencies
  */
-import { cloneElement } from '@wordpress/element';
+import { cloneElement, isValidElement } from '@wordpress/element';
 
-function Icon( { icon, size = 24, ...props } ) {
-	return cloneElement( icon, {
-		width: size,
-		height: size,
-		...props,
-	} );
+function Icon( { srcElement, size = 24, ...props } ) {
+	return (
+		isValidElement( srcElement ) &&
```

Comment:
`isValidElement` usage is good here, however, there's still no checks on what type of element is being used (i.e. `img` or `SVG`.  Is that okay?

Reply:
yeah propType would be good here.